### PR TITLE
'login'

### PR DESCRIPTION
**Pull request by RanaFathy123**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -24,7 +26,7 @@ const App = () => {
       errorElement: <NotFound />,
       children: [
         { index: true, element: <Login /> },
-        { path: "login", element: <Login /> },
+        { path: "login", element: <Login/> },
         { path: "register", element: <Register /> },
         { path: "forgetpass", element: <ForgetPassword /> },
```
Comment (Ahmed-Abou-Emran) [inline src/App.tsx:31]: The convention with naming URL paths is to make it small letters, and when having multiple words to separate  them with hyphen as the example below from Next.js page. 
image

The problem with using camelCase or having multiple words together is that search engines will treat these paths as a single word. If we try to select  it, we will find the following:
URL Naming

File: src/Modules/AuthModule/components/Login/Login.tsx [hunk 2]
```diff
@@ -1,6 +1,160 @@
...
+  // FormData : interface (Auth.ts)
+  const {
+    register,
+    handleSubmit,
+    formState: { errors},
+  } = useForm<FormData>();
+
+
+  // Password eye 
+  const [showPassword, setShowPassword] = useState<boolean>(false);
```
Comment (Ahmed-Abou-Emran) [inline src/Modules/AuthModule/components/Login/Login.tsx:25]: when we provide a default values, the type is inferred automatically by typescript. We can remove `<boolean>` here.

File: src/Modules/AuthModule/components/Login/Login.tsx [hunk 3]
```diff
@@ -1,6 +1,160 @@
...
+  const togglePasswordVisibility = (): void => {
+    setShowPassword((prevState: boolean) => !prevState);
+  };
+
+  // Send Data to Api
+  const onSubmit = async (data: FormData) => {
+
+    try {
+      const response = await axios.post(
+        "https://upskilling-egypt.com:3003/api/v1/Users/Login",
```
Comment (Ahmed-Abou-Emran) [inline src/Modules/AuthModule/components/Login/Login.tsx:36]: Having the URL hardcoded  like this is not a good practice, as this means that if we have the URL in 100 component and the URL has changed that we need to manually update all URLS.

An alternative approach to use axios.create to setup an axios instance that have the base URL and any other configuration, and when using this instance, we will pass only the relative path to the base URL.

We can also store the URL's in a constants  file like this one:
https://github.com/AnasKhaled7/PMS-React/blob/main/src/lib/APIs.ts

File: src/Modules/AuthModule/components/Login/Login.tsx [hunk 4]
```diff
@@ -1,6 +1,160 @@
...
+          
+            <div className="col-md-11 mt-3 ">
+
+              {/* Email */}
+              <div className="input m-3">
+          <label className="color-text" htmlFor="exampleFormControlInput1">
+            Email
+          </label>
+          <input
+            id="exampleFormControlInput1"
```
Comment (Ahmed-Abou-Emran) [inline src/Modules/AuthModule/components/Login/Login.tsx:75]: We can add meaning id name instead of this one.

**Comment by Ahmed-Abou-Emran** [comment]
image

We need to fix the autofill style as it doesn't match the design, When toggling the password as well, the input background turns to white. We need to handle that to match the design.

The login button's  vertical padding is a little bit big, we need to reduce it.

The focus style also need to be updated to match the design colors.
Also the toggle icon's not visible enough, we need to choose a color that has higher contrast ration than the existing one.
We also need to make  a horizontal padding, so that the text doesn't stick with the input edges.
image

**Comment by Ahmed-Abou-Emran** [comment]
The style of the error message doesn't look good. It also too big compared to the input with much padding. we need the message to take less portion from the design and the red color degree to match the design colors.
image

**Comment by Ahmed-Abou-Emran** [comment]
The design get broken when the error messages appear, as the error message make the form container bigger and the content overflows the image.
image

**Review by Ahmed-Abou-Emran — CHANGES_REQUESTED** [review]
Great job in the progress so far, I just left a few comments that suggest some improvements or fix some issues. You can look at them and request another review once you've finished.

If you have any question regarding any of the comments, you can ask me at any time.

File: src/App.tsx (hunk 1 above)
Comment (RanaFathy123) [reply src/App.tsx:31]: changed done

**Pull request merged**